src/App.test.js

```diff
@@ -13,11 +15,41 @@ const messages = {
   en: messages_en,
 };
 
-it('renders welcome message', () => {
-  const { getByText } = render(
-    <IntlProvider locale="en" messages={messages['en']}>
-      <App />
-    </IntlProvider>,
-  );
-  expect(getByText('Learn React')).toBeInTheDocument();
+const response = {
+  data: {
+    user: {
+      Email: 'fcoronel@makingsense.com',
+    },
+  },
+};
+
+describe('Test App component', () => {
+  afterEach(cleanup);
+
+  beforeEach(() => {
+    axios.get = jest.fn(() => Promise.resolve(response));
+  });
+
+  it('It fetches user and test welcome message', () => {
+    const { getByText } = render(
+      <IntlProvider locale="en" messages={messages['en']}>
+        <App />
+      </IntlProvider>,
+    );
+    expect(getByText('Learn React')).toBeInTheDocument();
+  });
+
+  it('It fetches user and display user data', async () => {
+    const { getByText } = render(
+      <IntlProvider locale="en" messages={messages['en']}>
+        <App />
+      </IntlProvider>,
+    );
+
+    await wait(() => getByText(response.data.user.Email));
```

Comment:
I am not sure about using DOM to verify when a request has been done to the backend. 

Although maybe it is fine, in that case, I would also verify if the request has been done.

Reply:
because use async and wait

Reply:
?

Reply:
With this code, we are telling the test to stop running the code inside of the test block and wait for the component to render the user returned from our API request. Once it renders that, it will continue down the list of assertions, but it will fail the test if the user isn’t rendered before the timeout (five second default).

Reply:
Yes, I understand, then we have talked, as I said, I would like to also test the request directly, not only through the DOM. 

But I do not know how to do it, so, I am fine with your test.